### PR TITLE
Server-Side game dimensions, mass rebalancing, SAT player collision, etc.

### PR DESCRIPTION
**Pull request by JoeMattie**
See commit notes for specifics on changes. This is a resubmission of my earlier pull request with more commit notes and has been brought up to date with upstream/master.

Sorry, I'm pretty new to git :/  Trying to get in the habit of committing more often

### CODE REVIEW

**Comment by JoeMattie** [comment]
Update: removed strict so Travis build would succeed

**Comment by neslinesli93** [comment]
Thanks for making commits more readable!
I think i've found a very weird bug in your code: when two players collide, the one who collided first immediately disconnects. I made these tests locally so I could easily say which of the players collided first since it was the only one on which I had control

**Comment by JoeMattie** [comment]
Fixed.  Was removing the user from the users array incorrectly

**Comment by JoeMattie** [comment]
Whups, closed by accident

**Comment by Buskcoin** [comment]
It seems too slow for me. I changed the log to base 10 and it's a lot more playable. I don't like the static background but it might be nice if you could toggle it. Check out my pr [#178](https://github.com/huytd/agar.io-clone/pull/178). With food generating around the player will the enemy players see more food around them as player aproaches?

**Comment by IgorAntun** [comment]
Thank you so much for that PR!
It seems like it has some merge conflicts though, would you be willing to fix those so I can merge it, please? :)

**Comment by JoeMattie** [comment]
Merge conflicts resolved

File: client/js/app.js [hunk 1]
```diff
@@ -465,13 +494,17 @@ function gameLoop() {
...
             
-            drawborder();
+            if(borderDraw){
+                drawborder();
+            }
             
+            foods.forEach(function(f){ drawFood(f); });
+    
+            if(borderDraw){
+                drawborder();
```
Comment (bigfoot90) [inline client/js/app.js]: Why you ```drawborder``` twice?

**Comment by Buskcoin** [comment]
I think that might be a mistake from merging 2 different versions of the drawBorder. It's not unque to this pr.

**Comment by IgorAntun** [comment]
Ooops, seems like other PR caused merge conflicts on this one again :\

**Comment by JoeMattie** [comment]
Fixed merge conflicts again

**Comment by IgorAntun** [comment]
Thank you so much for this! :)

**Pull request merged**